src/python_testing/TC_RVCCLEANM_2_2.py

```diff
@@ -0,0 +1,146 @@
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import chip.clusters as Clusters
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main, type_matches
+from mobly import asserts
+
+# This test requires several additional command line arguments
+# run with
+# --int-arg PIXIT_ENDPOINT:<endpoint> PIXIT.RVCCLEANM.MODE_CHANGE_OK:<mode id> PIXIT.RVCCLEANM.MODE_CHANGE_FAIL:<mode id>
+
+
+class TC_RVCCLEANM_2_2(MatterBaseTest):
+
+    def __init__(self, *args):
+        super().__init__(args)
+        self.endpoint = 0
+        self.supported_run_modes = {}  # these are the ModeOptionStructs
+        self.supported_run_modes_dut = []
+        self.supported_clean_modes_dut = []
+        self.run_mode_dut = 0
+        self.old_clean_mode_dut = 0
+        self.new_clean_mode_th = 0
+
+    async def read_mod_attribute_expect_success(self, cluster, attribute):
+        return await self.read_single_attribute_check_success(
+            endpoint=self.endpoint, cluster=cluster, attribute=attribute)
+
+    async def read_run_supported_modes(self) -> Clusters.Objects.RvcRunMode.Attributes.SupportedModes:
+        ret = await self.read_mod_attribute_expect_success(
+            Clusters.RvcRunMode,
+            Clusters.RvcRunMode.Attributes.SupportedModes)
+        asserts.assert_true(type_matches(ret, Clusters.Objects.RvcRunMode.Attributes.SupportedModes), "")
+        return ret
+
+    async def read_clean_supported_modes(self) -> Clusters.Objects.RvcCleanMode.Attributes.SupportedModes:
+        ret = await self.read_mod_attribute_expect_success(
+            Clusters.RvcCleanMode,
+            Clusters.RvcCleanMode.Attributes.SupportedModes)
+        asserts.assert_true(type_matches(ret, Clusters.Objects.RvcCleanMode.Attributes.SupportedModes), "")
+        return ret
+
+    async def send_change_to_mode_cmd(self, newMode) -> Clusters.Objects.RvcCleanMode.Commands.ChangeToModeResponse:
+        ret = await self.send_single_cmd(cmd=Clusters.Objects.RvcCleanMode.Commands.ChangeToMode(newMode=newMode), endpoint=self.endpoint)
+        asserts.assert_true(type_matches(ret, Clusters.Objects.RvcCleanMode.Commands.ChangeToModeResponse),
+                            "Unexpected return type for ChangeToMode")
+        return ret
+
+    # Prints the instruction and waits for a user input to continue
+    def print_instruction(self, step_number, instruction):
+        self.print_step(step_number, instruction)
+        input("Press Enter when done.\n")
+
+    @async_test_body
+    async def test_TC_RVCCLEANM_2_2(self):
+
+        asserts.assert_true('PIXIT_ENDPOINT' in self.matter_test_config.global_test_params,
```

Comment:
These tests now support --endpoint directly as a flag on the command line field - that's what the TH should be using, so you should use that.

self.endpoint = self.user_params.get("endpoint", 1)

(or omit the 1 if you don't want to include a default)

Reply:
I have tried this but the line that you suggested is not picking up the endpoint being set. `./scripts/tests/run_python_test.py --script src/python_testing/TC_RVCCLEANM_2_2.py --script-args "--storage-path admin_storage.json --PICS src/app/tests/suites/certification/ci-pics-values --endpoint 1"`

Am I missing something?

Reply:
fixed in latest revision.